### ISSUE TITLE
Test ConvertFromUriFormatAgnosticErrorsTest depends on the user's locale

### ISSUE THREAD

**Issue by zlatko-michailov** [OData/odata.net#428]
## Summary
The following variation of test __ConvertFromUriFormatAgnosticErrorsTest__ is locale-dependent:
```c#
                new ConvertFromUriLiteralTestCase()
                {
                    Parameter = "2011-02-04T00:00:00-08:00",
                    Model = edmModel,
                    TypeReference = edmModel.ResolveTypeReference("Edm.Double", false),
                    ExpectedException = ODataExpectedExceptions.ODataException("ODataUriUtils_ConvertFromUriLiteralTypeVerificationFailure", "Edm.Double", "2/4/2011 12:00:00 AM -08:00"),
                });
```

## Problem
The expected value "__2/4/2011 12:00:00 AM -08:00__" depends on a particular date/time format.

## Repro
* Open _Control Panel > Region_. 
* Change any of the following:
a) In the _Short date_ dropdown,  choose __yyyy-MM-dd__ (or any format different from the default US format.)
b) In the _Long time_ dropdown, choose, __HH:mm:ss__.
* Run the test.

You will see a failure like this:
```
Expected: Type verification failed. Expected type 'Edm.Double' but received the value '2/4/2011 12:00:00 AM -08:00'.
Actual:   Type verification failed. Expected type 'Edm.Double' but received the value '2011-02-04 00:00:00 -08:00'.
```

## Suggested Fix
Assign to me :wink:.

**Comment by congysu** [comment]
Thanks @zlatko-michailov.  I've sent an invitation to you.  Once you join the OData organization, I'll be able to select your GitHub ID from the assignee drop down list. :)

**Comment by zlatko-michailov** [comment]
Resolved with commit ca909f1.
